packages/winpanda/extra/src/winpanda/core/cmdconf.py

```diff
@@ -89,18 +90,68 @@ def __init__(self, **cmd_opts):
             # Make sure that the installation storage is in consistent state
             self.inst_storage.construct()
 
+        self._pkgrepopath = None
+        self._root_url = None
+
         # DC/OS cluster setup parameters
         self.cluster_conf_nop = False
-        self.cluster_conf = self.get_cluster_conf()
-        LOG.debug(f'{self.msg_src}: cluster_conf: {self.cluster_conf}')
+        self.dcosclusterpkginfopath = 'cluster-package-info.json'
+        self._cluster_conf = None
 
         # Reference list of DC/OS packages
         self.ref_pkg_list = self.get_ref_pkg_list()
         LOG.debug(f'{self.msg_src}: ref_pkg_list: {self.ref_pkg_list}')
 
         # DC/OS aggregated configuration object
-        self.dcos_conf = self.get_dcos_conf()
-        LOG.debug(f'{self.msg_src}: dcos_conf: {self.dcos_conf}')
+        self._dcos_conf = None
```

Comment:
This needs to appear before the call to `self.get_ref_pkg_list()` since it calls `get_cluster_conf` which tries to call `get_dcos_conf`.

Even if you make this change, I'm not sure that it won't cause further circular references.

Reply:
Added property dstor_pkglist_path to remove relations between `cluster_conf` and `get_ref_pkg_list` methods as I did this previously with `dcos_conf`